anipy_cli/mal.py

```diff
@@ -0,0 +1,613 @@
+        seasonal_list = seasonal.export_seasonals()
+        for anime in seasonal_list:
+            print(f"{colors.GREEN}Syncing {anime[0]} into MAL{colors.END} ...")
+            self.add_show(anime[0], anime[1], anime[2])
+            print(f"{colors.GREEN}Done.{colors.END}")
+
+    def auto_map_gogo_mal(self, mal_entry):
+        if "gogo_map" in mal_entry and len(mal_entry["gogo_map"]) > 0:
+            return True
+        print(
+            "{} Auto mapping:{} {}".format(
+                colors.GREEN, mal_entry["node"]["title"], colors.END
+            )
+        )
+
+        search_values = [
+            mal_entry["node"]["title"],
+            mal_entry["node"]["alternative_titles"]["en"],
+        ] + mal_entry["node"]["alternative_titles"]["synonyms"]
+
+        found = {}
+        for search in search_values:
+            query_class = query(search, entry)
+            query_class.get_pages()
+            found["search"] = query_class.get_links()
+
+            if found["search"] == 0:
+                self.shows_failed_automap.add(mal_entry["node"]["title"])
+                print(
+                    "{}{}: Failed.{}".format(
+                        colors.ERROR, mal_entry["node"]["title"], colors.END
+                    )
+                )
+                continue
+            if "gogo_map" not in mal_entry:
+                mal_entry["gogo_map"] = []
+            for i, anime in enumerate(found["search"][1]):
+                if any(
+                    anime.lower().rstrip("(dub)").rstrip("(japanese dub)").strip(" ")
+                    in show.lower()
+                    for show in [search_values[0], search_values[1]]
+                ):
+                    gogo_map = mal_entry["gogo_map"]
+                    current_map = self.make_gogo_map(found["search"][0][i], anime)
+
+                    self.update_gogo_map_list(gogo_map, current_map)
+                    self.shows_failed_automap.discard(mal_entry["node"]["title"])
+                    self.update_anime_list(
+                        mal_entry["node"]["id"],
+                        {
+                            "num_watched_episodes": mal_entry["node"]["my_list_status"][
+                                "num_episodes_watched"
+                            ],
+                            "tags": "anipy-cli",
+                        },
+                    )
+                else:
+                    self.shows_failed_automap.add(mal_entry["node"]["title"])
+        self.write_save_data()
+
+    def manual_map_gogo_mal(self, mal_anime_name: str, gogo: dict):
+        mal_entry = [
+            x
+            for x in self.local_mal_list_json["data"]
+            if x["node"]["title"] == mal_anime_name
+        ]
+        if len(mal_entry) > 0:
+            current_mal_entry = mal_entry[0]
+
+        else:
+            return False
+
+        if "gogo_map" not in current_mal_entry:
+            if not isinstance(current_mal_entry["gogo_map"], list):
```

Comment:
Here we get a keyerror when manual mapping, probably because `gogo_map` is not in the json yet.
```
Try auto mapping MAL to gogo format? (y/n):
n
MAL Anime that failed auto-map to gogo:
[1] Kaguya-sama wa Kokurasetai: Ultra Romantic
[2] Tate no Yuusha no Nariagari Season 2
[3] Kawaii dake ja Nai Shikimori-san
[4] Mushishi Zoku Shou
[5] Summertime Render
Selection: (e.g. 1, 1  3, 1-3 or * [for all] ) 
>> 1
Current: Kaguya-sama wa Kokurasetai: Ultra Romantic

[1] Kaguya-sama wa Kokurasetai: Ultra Romantic
[2] Kaguya-sama wa Kokurasetai: Ultra Romantic (Dub)
[3] Kaguya-sama wa Kokurasetai: Ultra Romantic Teaser PV - Ishigami Yuu wa Kataritai
[C]  Cancel.
Enter Number: 1
Traceback (most recent call last):
  File "/home/paul/Downloads/git/anipy-cli/./anipy_cli.py", line 14, in <module>
    main()
  File "/home/paul/Downloads/git/anipy-cli/./anipy_cli.py", line 8, in main
    cli.main()
  File "/home/paul/Downloads/git/anipy-cli/anipy_cli/cli.py", line 1029, in main
    mal_cli(
  File "/home/paul/Downloads/git/anipy-cli/anipy_cli/cli.py", line 649, in mal_cli
    m.take_input()
  File "/home/paul/Downloads/git/anipy-cli/anipy_cli/cli.py", line 700, in take_input
    self.create_gogo_maps()
  File "/home/paul/Downloads/git/anipy-cli/anipy_cli/cli.py", line 949, in create_gogo_maps
    self.m_class.manual_map_gogo_mal(
  File "/home/paul/Downloads/git/anipy-cli/anipy_cli/mal.py", line 601, in manual_map_gogo_mal
    current_mal_entry["gogo_map"],
KeyError: 'gogo_map'
```